src/qubes_miragevpn.md

```diff
@@ -0,0 +1,64 @@
+One advantage of MirageVPN is its ability to produce a [unikernel][mirageos],
+i.e. a fully-fledged operating system for a virtualizer such as [Xen]. In this
+respect, [QubesOS][qubesos] users can secure their VMs' Internet connections via
+a MirageVPN client.
+
+Unlike a standard OpenVPN client, the VM used is smaller and its attack surface
+just as reduced, as the operating system only does VPN: an OpenVPN client would
+require a kernel (like Linux), a kernel configuration to redirect VM connections
+to the VPN tunnel, and several libraries installed in order to function.
+
+In this chapter, we'll look at how to configure and install a MirageVPN client
+for QubesOS. We'll assume that the OpenVPN server is configured as described in
+[this handbook](./simple_openvpn_server.md).
+
+## Download and configuration
+
+You can download the unikernel from its official repository:
+https://github.com/robur-coop/qubes-miragevpn
+
+Next, copy the unikernel to dom0 with this command (from a dom0 terminal). In
+this example, we've uploaded the unikernel to the _app-vm_ `personal`:
+```sh
+$ mkdir -p /var/lib/qubes/vm-kernels/qubes-miragevpn/
+$ cd /var/lib/qubes/vm-kernels/qubes-miragevpn/
+$ qvm-run -p personal 'cat qubes-miragevpn.xen' > vmlinuz
+```
+
+Still from dom0, we now need to create a new VM from the downloaded image. An
+empty initramfs file must also be created:
+```sh
+$ gzip -n9 < /dev/null > initramfs
+$ qvm-create \
+  --property kernel=qubes-miragevpn \
+  --property kernelopts='' \
+  --property memory=256 \
```

Comment:
Qubes-mirage-firewall works with 32M, I suppose miragevpn can work with the same amount of RAM? To me the great point with unikernels is the low memory usage as well as the attack surface.

Reply:
With 32M, it does not live longer than 1h unfortunately 😕 .

Reply:
Ah, right, there is no `memory_pressure.ml` that triggers collection yet :)